docs/modules/ROOT/pages/reference/architecture.adoc

```diff
@@ -0,0 +1,170 @@
+inventory for any components that specify a version.
+The version specification is parsed from the inventory using Kapitan's
+inventory parsing, allowing the version to be overridden at any point in the
+configuration hierarchy.
+
+== Catalog Compilation
+
+Commodore uses https://kapitan.dev[Kaptian] to compile the cluster catalog.
+Kapitan is called with a few options enabled.
+Most importantly, Kapitan is configured to support fetching dependencies of
+components, such as Helm charts.
+Further, Kapitan is configured with an extended search path to support
+component libraries and the builtin `commodore.libjsonnet`.
+Finally, Kapitan is also configured to search for secret reference files in
+`catalog/refs` during compilation.
+See section <<_secrets_management>> for more details on the secrets management
+implemented with Commodore and Kapitan.
+
+=== Postprocessing filters
+
+After running Kapitan, Commodore applies postprocessing filters to the output
+of Kapitan.
+Postprocessing filters are defined in components and allow components to
+describe transformations that should be applied to the rendered manifests of
+the component.
+Commodore supports two types of postprocessing filters: _builtin_ filters and
+_jsonnet_ filters.
+Builtin filters are defined by Commodore itself.
+Commodore currently provides a single builtin filter `helm_namespace` which is
+intended to be used on files generated by the Kapitan helm plugin.
+A component can use the `helm_namespace` filter by providing the following
+filter configuration:
+
+.component-metrics-server/postprocess/filters.yml
+[source,yaml]
+--
+filters:
+  - path: metrics-server/01_helmchart/metrics-server/templates
+    type: builtin
+    filter: helm_namespace
+    filterargs:
+      namespace: ${metrics_server:namespace}
+      create_namespace: "true"
+--
+
+== Secrets Management
+
+Commodore makes use of https://kapitan.dev/secrets/[Kapitan's secrets
+management capabilities], but currently only supports references to secrets in
+Vault (called "Vaultkv" in the Kapitan documentation).
+
+Commodore takes care of generating secret reference files for any secret
+references (denoted by `?{vaultkv:...}`) found in key `parameters` in  all the
+classes included by the Kapitan cluster target.
+Secret references can use reclass references to define dynamic defaults, as
+Commodore searches for secret references in the rendered reclass inventory.
+
+Commodore saves the generated reference files are stored in the cluster
+catalog in directory `refs/`.
+This directory is configured as the base path in which Kapitan searches for
+reference files during compilation, allowing references in the inventory to
+omit the `catalog/refs` prefix which the would have to include otherwise.
+
+Because Commodore manages the secret files, it can guarantee that the secret
+files and the catalog are always in sync.
+All secret references MUST be made in the configuration parameters, otherwise
```

Comment:
We should probably also mention that the compiled output *must* include the secret ref in clear text (i.e. no base64), for example by using `stringData` for secrets.